Python_Engine/Compute/InstallPythonToolkit.cs

```diff
@@ -0,0 +1,79 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using System.IO;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Python
+{
+    public static partial class Compute
+    {
+        [Description("Installs the necessary pre-requisites to use the Python Toolkit fully. This will install Python 3.7 to your system, to the C:/ProgramData/BHoM folder.\n" +
+            "This can take several minutes to complete")]
+        [Input("run", "When you are ready to install Python, set this to true. Until this is set to true, this component will not run")]
+        [Input("force", "If you have previously installed a version of Python Toolkit, you may need to force the component to run a new install. This can happen if your system loses files. This is set to false by default, which means if we can detect a previous Python install on your system we will not install Python this time. Set this to true to ignore this check")]
+        [Output("success", "True is Python Toolkit has been successfully installer, false otherwise")]
+        public static bool InstallPythonToolkit(bool run = false, bool force = false)
+        {
+            if (!run)
+                return false;
+
+            // Install python
+            Console.WriteLine("Installing python 3.7 embedded...");
+            Compute.Install(force).Wait();
+
+            // Check the installation was successful 
+            if (!Query.IsInstalled())
+            {
+                BH.Engine.Reflection.Compute.RecordError("Coule not install Python");
+                return false;
+            }
+
+            // Install pip
+            Console.WriteLine("Installing pip...");
+            Compute.InstallPip();
+
+            // Check the pip installation was successful 
+            if (!Query.IsPipInstalled())
+            {
+                BH.Engine.Reflection.Compute.RecordError("Could not install pip");
+                return false;
+            }
+
+            // install project jupyter
+            Console.WriteLine("Installing jupyter...");
+            Compute.PipInstall("jupyter");
+            Compute.PipInstall("jupyterlab");
+            Compute.PipInstall("pythonnet");
+
```

Comment:
Maybe add `matplotlib` to the list? It should be
```
            Compute.PipInstall("matplotlib");
```

Reply:
In addition, would be nice to include multioutput called ```packagesInstalled``` or similar to show a list of everything after method completion for transparency.

Reply:
Sorry for being late to the party, I suspect we will need pandas, numpy, scipy when doing climate chart export, if this is not already included. 

Also looking at Tristan's WeatherChartGenerator.ipynb file, may need windrose, psychrometrics, ladybug, pysolar as well.

Reply:
Good point @BingWangUS - however I think we can tackle that one when @tg359 and I add the graph making script in the next PR if that's ok? Just to avoid this PR scope creeping into doing both bits of work :smile:

Reply:
Probably should have done the same for `matplotlib` thinking about it...

Reply:
Agree. 

Thinking further we should have a separate list of libraries used for the chart export (windrose, psychrometrics, ladybug, pysolar), which we can remove later. But pandas, scipy and numpy may need to go to the general list. (that is how I put into two paragraphs).

Reply:
Godd point @BingWangUS, but I suspect that those (numpy, scipy, pandas) would be responsibilities of the MachineLearning_Toolkit, which is where they are currently installed.

Reply:
This could be a separate discussion I think. Depending on how we split the responsibility of libraries, those numpy libraries could be a toolkit specified ones or general ones. If the MachineLearning is the only toolkit using python, as we said before, yes it should be in MachineLearning. If there is other application of python, such as currently what we are doing for the climate charts, it need be included in python toolkit. Understanding what we are doing now is a temporary solution for climate charts, I'm happy with either one but just sharing thoughts here.